ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Diffusion/Node/NodeKernel.hs

```diff
@@ -349,19 +357,21 @@ withNodeKernelThread
      , MonadThrow         m
      , MonadThrow    (STM m)
      , HasFullHeader block
+     , RandomGen seed
      )
   => BlockGeneratorArgs block seed
-  -> (NodeKernel header block m -> Async m Void -> m a)
+  -> (NodeKernel header block seed m -> Async m Void -> m a)
   -- ^ The continuation which has a handle to the chain selection \/ block
   -- production thread.  The thread might throw an exception.
   -> m a
 withNodeKernelThread BlockGeneratorArgs { bgaSlotDuration, bgaBlockGenerator, bgaSeed }
                      k = do
-    kernel <- newNodeKernel
+    let (_, psSeed) = split bgaSeed
```

Comment:
```suggestion
    let (bpSeed, psSeed) = split bgaSeed
```
Then `bpSeed` should be used below when passing to `loop` instead of `bgaSeed`.  Splitting an RNG is deterministic, so in the current implementation we might get correlations between two `rng`s that should be independent if it is split in the loop again.

Reply:
same as above